README.Rmd

```diff
@@ -49,136 +48,85 @@ pak::pak("tidyverse/glue")
 library(glue)
 
 name <- "Fred"
-glue('My name is {name}.')
-
-# A literal brace is inserted by using doubled braces.
-name <- "Fred"
-glue("My name is {name}, not {{name}}.")
+glue("My name is {name}.")
 ```
 
-`glue::glue()` is also made available via `stringr::str_glue()`.
-So if you've already attached stringr (or perhaps the whole tidyverse), you can access `glue()` like so:
+`stringr::str_glue()` is an alias for `glue::glue()`.
+So if you've already attached stringr (or perhaps the whole tidyverse), you can use `str_glue()` to access all of the functionality of `glue()`:
 
 ```{r eval = FALSE}
 library(stringr) # or library(tidyverse)
 
-stringr_fcn <- "`stringr::str_glue()`"
-glue_fcn    <- "`glue::glue()`"
-
-str_glue('{stringr_fcn} is essentially an alias for {glue_fcn}.')
-#> `stringr::str_glue()` is essentially an alias for `glue::glue()`.
+name <- "Wilma"
+str_glue("My name is {name}.")
+#> My name is Wilma.
 ```
 
-`glue_data()` works well with pipes:
+You're not limited to using a bare symbol inside `{}`; it can basically be any little bit of R code:
 
 ```{r}
-mtcars$model <- rownames(mtcars)
-mtcars |> head() |> glue_data("{model} has {hp} hp")
+name <- "Pebbles"
+glue("Here is my name in uppercase and doubled: {strrep(toupper(name), 2)}.")
 ```
 
-##### `glue_data()` is useful with [magrittr](https://cran.r-project.org/package=magrittr) pipes.
-```{r}
-`%>%` <- magrittr::`%>%`
-head(mtcars) %>% glue_data("{rownames(.)} has {hp} hp")
-```
+### Data can come from various sources
 
-##### `glue()` is useful within dplyr pipelines
-```{r, message = FALSE}
-library(dplyr)
-head(iris) %>%
-  mutate(description = glue("This {Species} has a petal length of {Petal.Length}"))
-```
+glue can interpolate values from the local environment or from data passed in `name = value` form:
```

Comment:
"Local environment" is a bit wrong/imprecise, but I don't want to be super pedantic here. Can you think of some thing much better?